File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.go
type OrderedStream struct {
    pointer int
    list []string
}


func Constructor(n int) OrderedStream {
    return OrderedStream{
        pointer : 0,
        list : make([]string, n),
    }
}


func (this *OrderedStream) Insert(idKey int, value string) []string {
    this.list[idKey-1] = value
    chunk := make([]string,0)
    for {
        if this.pointer == len(this.list) {
            break
        }
        if this.list[this.pointer] != "" {
            chunk = append(chunk,this.list[this.pointer])
            this.pointer++
        } else {
            break
        }
    }
    return chunk
}


/**
 * Your OrderedStream object will be instantiated and called as such:
 * obj := Constructor(n);
 * param_1 := obj.Insert(idKey,value);
 */

File: 1656-design-an-ordered-stream/1656-design-an-ordered-stream.java
class OrderedStream {

    String[] array;
    int pointer = 1;

    public OrderedStream(int n) {
        array = new String[n];
    }

    public List<String> insert(int idKey, String value) {
        array[idKey - 1] = value;
        List<String> answer = new ArrayList<>();
        while (pointer <= array.length && array[pointer - 1] != null) {
            answer.add(array[pointer - 1]);
            pointer++;
        }
        return answer;
    }
}

/**
 * Your OrderedStream object will be instantiated and called as such:
 * OrderedStream obj = new OrderedStream(n);
 * List<String> param_1 = obj.insert(idKey,value);
 */